Gas canister colors update

lessthnthree opened the pull request:
## About The Pull Request

- With the change of N2 to the gas color yellow in https://github.com/tgstation/tgstation/pull/78217, adjusts the canister to match.
- Empty/unlabelled are now grey, which is more fitting an empty canister.
- Since air mix and hydrogen are so similar, added red stripes to hydrogen.
- Antinob and halon had the same color, antinob changed to grey and yellow.

![image](https://github.com/tgstation/tgstation/assets/83487515/feab73ad-5b99-435f-afcb-3644266be032)

## Why It's Good For The Game

For the most part gas canisters match the color used in TGUI or the gas itself, for quick identification.

## Changelog

:cl: LT3
image: Nitrogen canisters are now yellow, antinob are grey/yellow, empty are grey, hydrogen are red/white
/:cl:

lessthnthree committed:
yellow N2 canister

lessthnthree committed:
Merge remote-tracking branch 'upstream/master' into nitrogen

lessthnthree committed:
Merge remote-tracking branch 'upstream/master' into nitrogen

lessthnthree committed:
Update canister.dm

lessthnthree committed:
red stripes on hydrogen

lessthnthree committed:
revert blue stripe of air

san7890 commented on code/modules/atmospherics/machinery/portable/canister.dm:
```diff
@@ -213,10 +213,10 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 	greyscale_colors = "#009823#f7d5d3"
 
 /obj/machinery/portable_atmospherics/canister/nitrogen
-	name = "Nitrogen canister"
-	gas_type = /datum/gas/nitrogen
-	greyscale_config = /datum/greyscale_config/canister
-	greyscale_colors = "#d41010"
```
why are they four-spaced instead of indented. did github crap out?

san7890 commented on code/modules/atmospherics/machinery/portable/canister.dm:
```diff
@@ -213,10 +213,10 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
-	gas_type = /datum/gas/nitrogen
-	greyscale_config = /datum/greyscale_config/canister
-	greyscale_colors = "#d41010"
+    name = "Nitrogen canister"
+    gas_type = /datum/gas/nitrogen
+    greyscale_config = /datum/greyscale_config/canister/double_stripe
+    greyscale_colors = "#e9ff5c#f4fce8"
```
why are the indents fourspacing instead of being tabs? is github crapping out

lessthnthree replied:
Good question

lessthnthree committed:
what

lessthnthree committed:
Merge branch 'master' into nitrogen

AtomTheProphet commented:
why does it say hydrogen has red stripes when they're clearly yellow?

MTandi commented:
Sorry, but these look like coder sprites.

This yellow has bad contrast with white.

The black hazard stripes blend with canister details on the empty one making it look like a clump of pixels.

That hydrogen canister looks like it's a broken sprite, the full red seems off.

Please ask for pallettes in sprinter discord.

Wallemations reviewed (changes requested):
desaturate the red stripes

lessthnthree committed:
desaturate hydrogen stripe, remove canister stripe to co2

lessthnthree commented:
Desaturated hydrogen, changed to stripe. Removed stripe from base canister, on CO2.

![image](https://github.com/tgstation/tgstation/assets/83487515/ccaba006-2dfa-4b30-ab5c-3e87f2c3adda)

Wallemations commented:
> > desaturate the red stripes
> 
> 
> 
> Desaturated hydrogen, changed to stripe. Removed stripe from base canister, on CO2.
> 
> 
> 
> ![image](https://github.com/tgstation/tgstation/assets/83487515/ccaba006-2dfa-4b30-ab5c-3e87f2c3adda)

Can you remind me which of these is which?

lessthnthree commented:
> > > desaturate the red stripes
> > 
> > 
> > 
> > Desaturated hydrogen, changed to stripe. Removed stripe from base canister, on CO2.
> > 
> > 
> > 
> > ![image](https://github.com/tgstation/tgstation/assets/83487515/ccaba006-2dfa-4b30-ab5c-3e87f2c3adda)
> 
> Can you remind me which of these is which?

L to R
Hydrogen, empty, nitrogen, CO2

Wallemations commented:
Doesn't water vapor also use a white stripe? Or is it a different color?

lessthnthree commented:
> Doesn't water vapor also use a white stripe? Or is it a different color?

Water vapor is like CO2 but beige instead of white

![image](https://github.com/tgstation/tgstation/assets/83487515/72ad50d0-47ab-4d2b-876b-4f4ee9c3ebbd)

lessthnthree commented:
Here's with changing the angry bees gas to dark grey and yellow.

![image](https://github.com/tgstation/tgstation/assets/83487515/f36d1000-8289-4023-8afc-8e623ad2cd51)

lessthnthree committed:
antinob grey and yellow

lessthnthree committed:
desaturate

san7890 commented:
can you update the name of this PR seeing as how it touches a lot more than just "yellow nitrogen"

Wallemations commented:
Honestly I still don't feel all that great about this touching so many canisters when you initially just went to make N2 yellow
Lots of people have muscle-memory with these so, idk

We could testmerge it and see how people feel I suppose

san7890 commented:
> Honestly I still don't feel all that great about this touching so many canisters when you initially just went to make N2 yellow
Lots of people have muscle-memory with these so, idk

I think it's okay to flesh this stuff out in the aftermath of the already-massive mega-shifting atmospherics gas visuals PR and just get it all right rather than prolong changes over a week. a test-merge will only really give a hundred or so players an advantage at most, i don't really see any value in it unless you specifically desire one for some reason.

nitrogen being yellow is the only really big change, seeing as antinob canisters will only exist when a user manually reskins it. hydrogen having red stripes isn't much a huge change either imo. empty and unlabelled being grey makes some level of intuitive sense. in an ideal world this would all be atomized but i think the non-yellow-N2 changes are benign enough.

the issue is the aforementioned PR has already sorta ripped the bandage off when it comes to gas color standardization, and it's probably more important to ensure the entire atmospherics framework is cohesive (since N2 is now yellow everywhere else after the TGUI stuff was updated).

lessthnthree commented:
Yeah, I don't want to touch any further because a full canister color coherency would be a much bigger thing. It's just like the antinob only showed up when I dropped all the cans, so it was a low hanging fruit.

MTandi commented:
Is it possible to combine stripes with hazard lines like so?

![image](https://github.com/tgstation/tgstation/assets/3625094/f27c2fb8-6ba5-4593-8aaa-45a664e1597d)

lessthnthree commented:
No, that would be adding an additional overlay. Improving the hazard stripe overlay is out of scope for this.

Wallemations commented:
> No, that would be adding an additional overlay. Improving the hazard stripe overlay is out of scope for this.

It's not entirely out of scope, all you'd really have to do is add both overlays at once with a new GAGS type
Hell I'm not entirely sure why we don't already have some extra mixings of the overlays, I kinda overlooked that when I last updated them

lessthnthree commented:
I like the black and yellow hazards, but if we're going to be introducing new stripe types it should be part of a coordinated redo of all the canister color/stripes.

Wallemations commented:
> I like the black and yellow hazards, but if we're going to be introducing new stripe types it should be part of a coordinated redo of all the canister color/stripes.

which is more or less what you're already doing

lessthnthree commented:
Then it can all wait until all of the canisters get a pass over someday.

Wallemations commented:
I don't see why you closed it, I never said I wanted you to do it, just that it would make sense if you did, but alright

lessthnthree commented:
> I don't see why you closed it, I never said I wanted you to do it, just that it would make sense if you did, but alright

It seemed like that was desired, and I didn't want to do any sprite work right now, like changing the hazards.

If a consensus wants just N2, the four cans, whichever it's fine with me, I offered to atomize.

I don't however want to keep changing these and going through the screenshots again, etc.

Toastgoats commented:
Why the fuck is N2 piss yellow? I know its supposedly the color of N2 irl but last I checked N2O isn't brick red.
Just let it be red but put a white center stripe to imply that it's a counterpart of oxygen.
